framework/include/utils/MooseMeshUtils.h

```diff
@@ -289,4 +289,21 @@ void makeOrderedNodeList(std::vector<std::pair<dof_id_type, dof_id_type>> & node
                          std::vector<dof_id_type> & elem_id_list,
                          std::vector<dof_id_type> & ordered_node_list,
                          std::vector<dof_id_type> & ordered_elem_id_list);
+
+/**
+ * Converts a given name (BoundaryName or SubdomainName) into a corresponding ID (BoundaryID or
+ * SubdomainID) and performs bounds checking to ensure that overflow doesn't happen.
+ * @param name Name that is to be converted into an ID.
+ * @return ID type corresponding to the type of name.
+ */
+template <typename T, typename Q>
+Q
+getIDFromName(const T & name)
+{
+  const auto id = stoi(name);
+  if (id < std::numeric_limits<Q>::min() || id > std::numeric_limits<Q>::max())
+    mooseError(name, " is not within the numeric limits of the expected ID type.");
```

Comment:
specify the type in the error message with prettyCppName<Q>()

Reply:
Where is prettyCppName defined? I don't see it in MOOSE or libmesh, and google gives me nothing.

Reply:
he means prettyCppType

Reply:
prettyCppType demangles `BoundaryName` correctly, but demangles `BoundaryID` as `short`, which is not desirable.

Reply:
Well 'short' is googlable for its size unlike BoundaryID so it may not be so bad?

Reply:
Yeah, I still think it's better than not putting the type.